### DISCUSSION TITLE
Trouble rendering some unicode characters

### DISCUSSION THREAD

**Discussion by rcorre** [bevyengine/bevy#13474] [Q&A] [answered]
I'm using https://sourcefoundry.org/hack/ as my font, which is also the font I use in my terminal ([st](https://dwm.suckless.org/)). When used in bevy, I've noticed it fails to render some unicode characters which render fine in my terminal.

Game:
![1716389281](https://github.com/bevyengine/bevy/assets/2496231/507ec098-eb86-423c-a9a3-b7ebe9959007)

Terminal:
![1716389312](https://github.com/bevyengine/bevy/assets/2496231/6257edcd-cb00-4e6b-8556-8f430e3d3ac6)

The offending character is ␣, (U+2423). Any idea why this would render in my terminal (and other applications where I use the same font), but not bevy?

**Comment by pkupper** [comment, marked as answer]
Are you sure that character is included in the font you are using? Your terminal probably has some type of font fallback implemented which Bevy's UI does not. Font fallback will likely come to Bevy with the planned migration to cosmic-text.

**Reply by rcorre** [reply]
You're correct, I think that's exactly what fooled me. Thanks!
